File: 프로그래머스/1/468371. 노란불 신호등/노란불 신호등.java
class Solution {
    public int solution(int[][] signals) {
        int answer = 0;
        
        while (answer<5000000){
            answer+=1;
            boolean all_yellow = true;
            for (int i=0;i<signals.length;i++){
                int G=signals[i][0];
                int Y=signals[i][1];
                int R=signals[i][2];
                int tmp=answer%(G+Y+R);
                if (G >= tmp || tmp> G + Y){
                    all_yellow = false;
                    break;
                }
                
            }
                
            if (all_yellow){
                return answer;
            }
            
            
        }
        
    
        return -1;
    }
}

File: 프로그래머스/1/468371. 노란불 신호등/노란불 신호등.py
def solution(signals):    
    answer = 0

    while answer<5000000 :
        answer+=1
        all_yellow = True
        for i in range(len(signals)):
            G=signals[i][0]
            Y=signals[i][1]
            R=signals[i][2]
            tmp=answer%(G+Y+R)
            
            if not (G < tmp <= G + Y):
                all_yellow = False
                break
        if all_yellow:
            return answer
    return -1